File: CS/dxSample/Form1.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Windows.Forms;
using DevExpress.DataAccess.ConnectionParameters;
using DevExpress.DataAccess.Sql;
using DevExpress.XtraGrid;
using DevExpress.XtraGrid.Views.Grid;

namespace dxSample {
    public partial class Form1 : DevExpress.XtraEditors.XtraForm {
        private GridControl _Grid;
        public GridControl Grid {
            get {
                if (_Grid == null) {
                    CreateGridControl();
                }
                return _Grid;
            }
        }
        public Form1() {
            InitializeComponent();
        }
        private void Form1_Load(object sender, EventArgs e) {
            Init();
        }
        private void Init() {
            SqlDataSource ds = InitData();
            Grid.DataSource = ds;
            Grid.DataMember = "customQuery1";
        }
        private void CreateGridControl() {
            _Grid = new GridControl();
            GridView view = new GridView();
            _Grid.ViewCollection.Add(view);
            _Grid.MainView = view;
            _Grid.Dock = DockStyle.Fill;
            Controls.Add(_Grid);
        }
        private static SqlDataSource InitData() {
            MsSqlConnectionParameters connectionParameters = new MsSqlConnectionParameters("localhost", "nwind.mdf", "username", "password", MsSqlAuthorizationType.SqlServer);
            SqlDataSource ds = new SqlDataSource(connectionParameters);
            CustomSqlQuery query = new CustomSqlQuery();
            query.Name = "customQuery1";
            query.Sql = "SELECT * FROM Products";
            ds.Queries.Add(query);
            ds.Fill();
            return ds;
        }
    }
}

File: VB/dxSample/Form1.vb
Imports System
Imports System.Windows.Forms
Imports DevExpress.DataAccess.ConnectionParameters
Imports DevExpress.DataAccess.Sql
Imports DevExpress.XtraGrid
Imports DevExpress.XtraGrid.Views.Grid

Namespace dxSample

    Public Partial Class Form1
        Inherits DevExpress.XtraEditors.XtraForm

        Private _Grid As GridControl

        Public ReadOnly Property Grid As GridControl
            Get
                If _Grid Is Nothing Then
                    CreateGridControl()
                End If

                Return _Grid
            End Get
        End Property

        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs)
            Init()
        End Sub

        Private Sub Init()
            Dim ds As SqlDataSource = InitData()
            Grid.DataSource = ds
            Grid.DataMember = "customQuery1"
        End Sub

        Private Sub CreateGridControl()
            _Grid = New GridControl()
            Dim view As GridView = New GridView()
            _Grid.ViewCollection.Add(view)
            _Grid.MainView = view
            _Grid.Dock = DockStyle.Fill
            Controls.Add(_Grid)
        End Sub

        Private Shared Function InitData() As SqlDataSource
            Dim connectionParameters As MsSqlConnectionParameters = New MsSqlConnectionParameters("localhost", "nwind.mdf", "username", "password", MsSqlAuthorizationType.SqlServer)
            Dim ds As SqlDataSource = New SqlDataSource(connectionParameters)
            Dim query As CustomSqlQuery = New CustomSqlQuery()
            query.Name = "customQuery1"
            query.Sql = "SELECT * FROM Products"
            ds.Queries.Add(query)
            ds.Fill()
            Return ds
        End Function
    End Class
End Namespace
